### ISSUE TITLE
Error: "ValueError: 'z0' is not in list" after adding 5th joint

### ISSUE THREAD

**Issue by alhinai21** [LinuxCNC/linuxcnc#637]
## Here are the steps I follow to reproduce the issue:

1. Configure a new mechine using Pncconf and then add 2 more joints in Z and Y axis as the machine have 3 axis and 5 joints (XYYZZ).

[alhinai_plasma2.hal.txt](https://github.com/LinuxCNC/linuxcnc/files/3549324/alhinai_plasma2.hal.txt)
[alhinai_plasma2.ini.txt](https://github.com/LinuxCNC/linuxcnc/files/3549325/alhinai_plasma2.ini.txt)

2. Changing gmoccapy to axis with same ini and hal files will make it run without issues.

3. After adding 5th joint (X) to example confige file _sim.gmoccapy.non_trivial_kinematics.Gantry_ same issue appear except it say y0 instead of z0

## This is what I expected to happen:
Run without any issues.
## This is what happened instead:
```
Traceback (most recent call last):
  File "/usr/bin/gmoccapy", line 5158, in <module>
    app = gmoccapy(sys.argv)
  File "/usr/bin/gmoccapy", line 281, in __init__
    self._make_DRO()
  File "/usr/bin/gmoccapy", line 566, in _make_DRO
    joint = self._get_joint_from_joint_axis_dic(axis)
  File "/usr/bin/gmoccapy", line 589, in _get_joint_from_joint_axis_dic
    return self.joint_axis_dic.keys()[self.joint_axis_dic.values().index(value)]
ValueError: 'z0' is not in list
```

## It worked properly before this:
after upgrading from https://github.com/phillc54/linuxcnc-plasmac to 2.8 or 2.9

## Information about my hardware and software:

 * I am using this Linux distribution and version: Debian GNU/Linux 9.9 (stretch)
 * I am using this kernel version:Linux plasma 4.9.0-9-rt-amd64 #1 SMP PREEMPT RT Debian 4.9.168-1+deb9u5 (2019-08-11) x86_64 GNU/Linux
 * I am running ...
   * [x] A binary version from linuxcnc.org (including buildbot.linuxcnc.org)
   * [ ] A binary I built myself
   * [ ] A binary version from some other source besides linuxcnc.org
 * I am using this LinuxCNC version:
1- linuxcnc 2.8.0-pre1-5093-ga042b0053
2- linuxcnc 2.9.0-pre0-507-g0dbddad61
 * I am using this user interface (GUI):gmoccapy
 * I am using this interface hardware vendor and chipset (e.g., parallel port, ethernet port, FPGA card):  MESA 5i25 + 7i76

**Comment by alhinai21** [comment]
check out to e2928c43f8e27b8d7a84b84e34747a10d8c08c4c will solve the problem, I think bed140fb8695b4a40aa13dd68ab33ea62b99882d cause it

**Comment by ra7bi** [comment]
same problem here , Pls help

**Comment by kd-kraken** [comment]
I've found a solution to this issue but don't know anything about using GitHub to submit code updated. I've posted it here https://forum.linuxcnc.org/gmoccapy/38201-gmoccapy-won-t-load-on-fresh-config

**Comment by gmoccapy** [comment]
Fixed that with cea561b on 11.10.2020 - 19:59

thanks to AgentWD40 for the provided code, I used that with one minor change

**Comment by gmoccapy** [comment]
Fixed that with cea561b on 11.10.2020 - 19:59

thanks to AgentWD40 for the provided code, I used that with one minor change

**Comment by gmoccapy** [comment]
Fixed that with cea561b on 11.10.2020 - 19:59

thanks to AgentWD40 for the provided code, I used that with one minor change
